File: myshoppinglistapi/src/main/java/com/example/myshoppinglistapi/services/UserService.java
package com.example.myshoppinglistapi.services;

import com.example.myshoppinglistapi.entities.User;
import com.example.myshoppinglistapi.exceptions.ObjectNotFoundException;
import com.example.myshoppinglistapi.exceptions.UserExistException;
import com.example.myshoppinglistapi.repositories.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {
    @Autowired
    private UserRepository userRepository;

    public List<User> findAllUser(){
        return userRepository.findAll();
    }

    public User findById(String email, String password) throws ObjectNotFoundException {
        Optional<User> user = userRepository.findByEmailAndPassword(email, password);

        if(user.isPresent()) {
            return user.get();
        } else {
            throw new ObjectNotFoundException("Usuário não encontrado! email: " + email);
        }
    }

    public User saveUser(User user) throws UserExistException {
        try{
            findById(user.getEmail(), user.getPassword());
            throw new UserExistException("Usuário já existe!");
        }catch (ObjectNotFoundException objectNotFoundException){
            return userRepository.save(user);
        }
    }

    public User updateUser(User user) throws ObjectNotFoundException {
        try{
            User userOld = findById(user.getEmail(), user.getPassword());

            userOld.setName(user.getName());
            userOld.setNickName(user.getNickName());
            userOld.setIdAvatar(user.getIdAvatar());

            return userRepository.save(userOld);
        }catch (ObjectNotFoundException objectNotFoundException){
            throw objectNotFoundException;
        }
    }
}

File: myshoppinglistapi/src/main/kotlin/com/example/myshoppinglistapi/services/UserService.kt
package com.example.myshoppinglistapi.services

import com.example.myshoppinglistapi.entities.User
import com.example.myshoppinglistapi.exceptions.ObjectNotFoundException
import com.example.myshoppinglistapi.exceptions.UserExistException
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import com.example.myshoppinglistapi.repositories.UserRepository
import java.lang.Exception
import java.util.*

@Service
class UserService() {

    @Autowired
    lateinit var userRepository: UserRepository

    fun findAllUser(): List<User>{
        return userRepository.findAll()
    }

    fun findById(email: String, password: String) : User {
        val user = userRepository.findByEmailAndPassword(email, password)
        return if(user.isPresent) user.get() else throw ObjectNotFoundException("Usuário não encontrado! email: ${email}")
    }

    fun saveUser(user: User): User{
        try{
            findById(user.email, user.password)
            throw UserExistException("Usuário já existe!")
        }catch (objectNotFoundException: ObjectNotFoundException){
            return userRepository.save(user)
        }
    }

    fun updateUser(user: User): User{
        try {
            val userOld = findById(user.email, user.password)

            userOld.name = user.name
            userOld.nickName = user.nickName
            userOld.idAvatar = user.idAvatar

            return userRepository.save(userOld)
        }catch  (objectNotFoundException: ObjectNotFoundException){
            throw objectNotFoundException
        }
    }
}